mapstory/import_handlers.py

```diff
@@ -51,3 +53,63 @@ def handle(self, layer, layer_config, *args, **kwargs):
                 for date_option in ('start_date', 'end_date'):
                     if layer_config.get(date_option) == date_field:
                         layer_config[date_option] = new_field_name.lower()
+
+
+class GeoserverHandlerMixin(ImportHandlerMixin):
+    """
+    A Mixin for Geoserver handlers.
+    """
+    catalog = gs_catalog
+
+
+class DefaultStyleHandler(GeoserverHandlerMixin):
+    """
+    Apply a default MapStory theme SLD style to imported layer
+    """
+
+    def can_run(self, layer, layer_config, *args, **kwargs):
+        """
+        Only run this handler if the layer is found in Geoserver and the layer's style is one of the default styles.
+        """
+        self.catalog._cache.clear()
+        self.layer = self.catalog.get_layer(layer)
+
+        if self.layer and self.layer.default_style:
+            return self.layer.default_style.name in ['polygon', 'point', 'line']
+        else:
+            return False
+
+    @ensure_can_run
+    def handle(self, layer, layer_config, *args, **kwargs):
+        """
+        Default MapStory style applied depending on feature type
+        """
+
+        # import pydevd
+        # pydevd.settrace('192.168.0.15', port=65432, stdoutToServer=True, stderrToServer=True, trace_only_current_thread=False,overwrite_prev_trace=True,patch_multiprocessing=True)
+
+        sld = self.get_feature_type(self.layer)
+
+        layer = self.catalog.get_layer(self.layer.name)
```

Comment:
layer is already included as an arg to this method

Reply:
renamed variable to make more clear